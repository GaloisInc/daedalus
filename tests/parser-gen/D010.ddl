-- testing input synchronization during lockstep move

def Chunk n P =  {
  @cur  = GetStream;
  @this = Take n cur;
  @next = Drop n cur;
  SetStream this;
  $$ = P;
  SetStream next;
}

def CanDisambiguate =
  Choose
  { x = Chunk 2 { Match "ab" }
  ; y = Chunk 2 { Match "ac" }
  }

def ShouldAbortWithIncompatibleInput =
 Choose
 { x = { Chunk 1 { Match "a" }
       ; Match "b"
       }
 ; y = Chunk 2 { Match "ac" }
 }

def Main =
  { @a = CanDisambiguate
  ; @b = ShouldAbortWithIncompatibleInput
  ; END
  }
