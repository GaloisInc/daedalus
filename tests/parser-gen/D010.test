dump
D010.ddl
--gen
