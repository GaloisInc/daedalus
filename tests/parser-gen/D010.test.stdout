**********************************
***** Strict LLA Transitions *****
**********************************
Stack:
    *
    Main callsite __START__
Start: Main (28,10)-(28,41)
SlkCfg{ q:180, ctrl:[?, *], sem:[?, ], inp:---[2..] }
AbortDFAIncompatibleInput


Stack:
    *
Start: Main (26,1)-(30,3)
SlkCfg{ q:1, ctrl:[*], sem:[*], inp:--- }
DTrans [
  ( [(38),(38),]
  , HeadInput ['a']
  , DTrans [
      ( [(26),]
      , HeadInput ['b']
      , Resolution (NotAmbiguous)
      ),
      ( [(26),]
      , HeadInput ['c']
      , Resolution (NotAmbiguous)
      ),
    ]
  ),
]

**********************
**** Extended LLA ****
**********************
("AbortDFAIncompatibleInput",8)
("DataDepTrans",11)
("Lookahead-0",2)
("Lookahead-1",11)
("Lookahead-2",3)

Total nb states: 35

**********************
***** Strict LLA *****
**********************
("AbortDFAIncompatibleInput",1)
("Lookahead-2",1)

Total nb DFAs: 2


Warning: LL(*) failures:

Stack:
    *
    Main callsite __START__
Start: Main (28,10)-(28,41)
SlkCfg{ q:180, ctrl:[?, *], sem:[?, ], inp:---[2..] }
AbortDFAIncompatibleInput


